src/X86Optimize.cpp

```diff
@@ -0,0 +1,633 @@
+#include "X86Optimize.h"
+
+#include "CSE.h"
+#include "FindIntrinsics.h"
+#include "IR.h"
+#include "IRMatch.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "InstructionSelector.h"
+#include "Simplify.h"
+
+namespace Halide {
+namespace Internal {
+
+#if defined(WITH_X86)
```

Comment:
I don't think this is necessary. It's only necessary if the file is going to refer to llvm x86-specific stuff